### DISCUSSION TITLE
Warp tool and keyframes

### DISCUSSION THREAD

**Discussion by inTIMidation297** [keiwando/toonsquid#507] [Q&A]
Is there a way to use the warp tool and make it a keyframe? Like how you can change the size and rotation.

**Comment by Cat-Butler** [comment]
Not currently. It's been requested a few times and might possibly be considered in the future (see https://github.com/keiwando/toonsquid/discussions/50 ). 

Currently, you can keep duplicating a warped drawing and modify it in a frame-by-frame approach, although that would start with a fresh warp mesh on each new drawing.

**Reply by HyperAnimated** [reply]
Making sure I understand, please…say we have a drawing, we use perspective or warp on it, and all we want to do is keyframe tween the drawing’s position/scale from there.  Is that impossible?

I can’t tell if people are just asking to tween the warp effect itself or if warping makes all tweening impossible.

Thanks for your help!

**Reply by Cat-Butler** [reply]
We're discussing the former - tweening the warp effect itself with keyframes, which would allow for e.g. some useful organic facial animation, amongst other uses. Not currently a feature.

A drawing with warp or perspective applied can definitely be tweened/keyframed afterwards with position/scale/rotation in the current version.

**Comment by HyperAnimated** [comment]
Thanks for responding!  And weird, I tried it for the first time this weekend (newbie) and after using warp or perspective, it ignored anything else I tried to do with keyframes.

Even if I, say, manually set keyframes for start and end position, then moved the drawing, no tween appeared, the drawing was just static at the end position.  Is there a step I’m missing?

**Reply by keiwando** [reply]
This, probably: https://toonsquid.com/handbook/guides/faq/#pixel-layer-keyframes

**Reply by HyperAnimated** [reply]
Ack, sorry for the slow response!  That fixed it, thank you!
